### PR TITLE
Add MLFLOW_S3_ENDPOINT_URL to specify custom S3 endpoint

### PR DESCRIPTION
**Pull request by hamroune**
add _get_s3_client method

### CODE REVIEW

**Comment by smurching** [comment]
Thanks for the updates! Just a heads up, you can update the code in an existing PR by pushing to your PR branch (for example, pushing to branch "450" of this PR will update the code in this PR & retrigger Travis tests)

File: mlflow/store/s3_artifact_repo.py [hunk 1]
```diff
@@ -52,7 +56,9 @@ def list_artifacts(self, path=None):
             dest_path = build_path(dest_path, path)
         infos = []
         prefix = dest_path + "/"
-        paginator = boto3.client('s3').get_paginator("list_objects_v2")
+        s3_endpoint_url = os.environ.get('MLFLOW_S3_ENDPOINT_URL')
```
Comment (smurching) [inline mlflow/store/s3_artifact_repo.py]: Could we also call `self._get_s3_client` here?

**Review by smurching — APPROVED** [review]
Just one small comment otherwise LGTM!

**Comment by hamroune** [comment]
I close the previous PR ad correct this

**Review by smurching — APPROVED** [review]
LGTM! Did you get a chance to manually test these changes (e.g. by reading/writing from a minios server)?

**Comment by hamroune** [comment]
yes tested locally with local minio server

start mlflow and point s3 endpoint to minio
image

we can see nothing in mlflow 
image

also nothing yet in minio
image

when execute train
image

I can see the result in mlflow
image

and as expected in minio
image

**Pull request merged**